OpenRA.Mods.Common/AI/States/NavyStates.cs

```diff
@@ -0,0 +1,194 @@
+	{
+		public void Activate(Squad owner) { }
+
+		public void Tick(Squad owner)
+		{
+			if (!owner.IsValid)
+				return;
+
+			if (!owner.IsTargetValid)
+			{
+				var t = FindClosestEnemy(owner);
+				if (t != null)
+					owner.TargetActor = t;
+				else
+				{
+					owner.FuzzyStateMachine.ChangeState(owner, new NavyUnitsFleeState(), true);
+					return;
+				}
+			}
+
+			var leader = owner.Units.ClosestTo(owner.TargetActor.CenterPosition);
+			if (leader == null)
+				return;
+			var ownUnits = owner.World.FindActorsInCircle(leader.CenterPosition, WDist.FromCells(owner.Units.Count) / 3)
+				.Where(a => a.Owner == leader.Owner && owner.Units.Contains(a)).ToHashSet();
+			if (ownUnits.Count < owner.Units.Count)
+			{
+				owner.Bot.QueueOrder(new Order("Stop", leader, false));
+				foreach (var unit in owner.Units.Where(a => !ownUnits.Contains(a)))
+					owner.Bot.QueueOrder(new Order("AttackMove", unit, false) { TargetLocation = leader.Location });
+			}
+			else
+			{
+				var enemies = owner.World.FindActorsInCircle(leader.CenterPosition, WDist.FromCells(12))
+					.Where(a1 => !a1.Disposed && !a1.IsDead);
+				var enemynearby = enemies.Where(a1 => a1.Info.HasTraitInfo<ITargetableInfo>() && leader.Owner.Stances[a1.Owner] == Stance.Enemy);
+				var target = enemynearby.ClosestTo(leader.CenterPosition);
+				if (target != null)
+				{
+					owner.TargetActor = target;
+					owner.FuzzyStateMachine.ChangeState(owner, new NavyUnitsAttackState(), true);
+					return;
+				}
+				else
+					foreach (var a in owner.Units)
+						owner.Bot.QueueOrder(new Order("AttackMove", a, false) { TargetLocation = owner.TargetActor.Location });
+			}
+
+			if (ShouldFlee(owner))
+			{
+				owner.FuzzyStateMachine.ChangeState(owner, new NavyUnitsFleeState(), true);
+				return;
+			}
+		}
+
+		public void Deactivate(Squad owner) { }
+	}
+
+	class NavyUnitsAttackState : NavyStateBase, IState
+	{
+		public void Activate(Squad owner) { }
+
+		public void Tick(Squad owner)
+		{
+			if (!owner.IsValid)
+				return;
+
+			if (!owner.IsTargetValid)
+			{
+				var t = FindClosestEnemy(owner);
+				if (t != null)
+					owner.TargetActor = t;
+				else
+				{
+					owner.FuzzyStateMachine.ChangeState(owner, new NavyUnitsFleeState(), true);
+					return;
+				}
+			}
+
+			// Switch target durign fight
+			var targetActor = owner.Bot.FindClosestEnemy(owner.Units.First().CenterPosition);
+			foreach (var a in owner.Units)
+				if (!BusyAttack(a))
+					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = targetActor });
+
+			if (ShouldFlee(owner))
+			{
+				owner.FuzzyStateMachine.ChangeState(owner, new NavyUnitsFleeState(), true);
+				return;
+			}
+		}
+
+		public void Deactivate(Squad owner) { }
+	}
+
+	class NavyUnitsFleeState : NavyStateBase, IState
+	{
+		public void Activate(Squad owner) { }
+
+		public void Tick(Squad owner)
+		{
+			if (!owner.IsValid)
+				return;
+
+			GoToRandomOwnBuilding(owner);
```

Comment:
Yes, I think it is the best way to make the naval units flee when the SPEN/SYRD is gone.